src/main/java/doug/command/Doug.java

```diff
@@ -0,0 +1,420 @@
+package doug.command;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import doug.tasks.*;
+import  java.io.File;
+import  java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+
+public class Doug {
+
+    // Declaring Magic Literal
+    private static final String DASHED_LINE = "______________________________________________\n";
+
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private static int counter = 0;
+    private static boolean saidBye = false;
+
+    public static void checkOutOfBounds(int listIndex) throws DougException {
+        if (listIndex > counter || listIndex <= 0) {
+            throw new DougException(DASHED_LINE + "No can do bud, your input is invalid!\n" + DASHED_LINE);
+        }
+    }
+
+    public static void doList() {
+        if (counter == 0) {
+            System.out.println(DASHED_LINE + "Got nothing on your roster bud.");
+            System.out.print(DASHED_LINE);
+            return;
+        }
+
+        System.out.println(DASHED_LINE + "Here, let me lay out your tasks for you.");
+        for (int i = 0; i < counter; i++) {
+            System.out.println((i+1) + "." + tasks.get(i).toString());
+        }
+        System.out.print(DASHED_LINE);
+    }
+
+    public static void doMark(String command) throws DougException {
```

Comment:
Naming of methods could be clearer. Using 2 verbs isn't very informative. You are "do"-ing "mark"-ing, but marking what?